packages/core/src/shareback/ensure-ccd-exists.ts

```diff
@@ -0,0 +1,43 @@
+import { executeWithNetworkRetries } from "@metriport/shared";
+import axios from "axios";
+import { S3Utils } from "../external/aws/s3";
+import { Config } from "../util/config";
+import { CCD_SUFFIX, createUploadFilePath } from "../domain/document/upload";
+
+const apiUrl = Config.getApiUrl();
+const region = Config.getAWSRegion();
+const s3Utils = new S3Utils(region);
+const api = axios.create();
+const bucket = Config.getMedicalDocumentsBucketName();
+
+export async function ensureCcdExists({
+  cxId,
+  patientId,
+  log,
+}: {
+  cxId: string;
+  patientId: string;
+  log: typeof console.log;
+}): Promise<void> {
+  const destinationKey = createUploadFilePath(cxId, patientId, `${CCD_SUFFIX}.xml`);
+  const ccdExists = await s3Utils.fileExists(bucket, destinationKey);
+  if (ccdExists) return;
+
+  log("No CCD found. Let's trigger generating one.");
+  const queryParams = {
+    cxId,
+    patientId,
+  };
+  const params = new URLSearchParams(queryParams).toString();
+
+  executeWithNetworkRetries(async () => api.post(`${apiUrl}/internal/docs/ccd?${params}`), {
+    log,
+  });
+  await executeWithNetworkRetries(
+    async () => await api.post(`${apiUrl}/internal/docs/empty-ccd?${params}`),
+    { log }
+  );
```

Comment:
Looking at these, wouldn't it be better to first create the empty CCD and then trigger generating the new one?

Just in case, in the unlikely event that `POST /internal/docs/ccd` runs super fast, we don't override a good CCD containing data w/ an empty one. 😄 

I know this is how it was, but I was already wanting to talk about this on the last review.

Reply:
💯 we should do that